test/dattri/algorithm/test_rps.py

```diff
@@ -0,0 +1,117 @@
+"""Test for RPS."""
+
+import torch
+from torch import nn
+from torch.nn import functional
+from torch.utils.data import DataLoader, TensorDataset
+
+from dattri.algorithm.rps import RPSAttributor
+from dattri.benchmark.utils import SubsetSampler
+
+
+class MLP(nn.Module):
+    """A multi-layer perceptron model for MNIST dataset."""
+
+    def __init__(
+        self,
+        input_size=28 * 28,
+        hidden_size1=128,
+        hidden_size2=64,
+        num_classes=10,
+    ):
+        """Initialize the multi-layer perceptron model."""
+        super(MLP, self).__init__()
+        self.fc1 = nn.Linear(input_size, hidden_size1)
+        self.fc2 = nn.Linear(hidden_size1, hidden_size2)
+        self.fc3 = nn.Linear(hidden_size2, num_classes)
+
+    def forward(self, x):
+        """Forward pass of the multi-layer perceptron model."""
+        x = x.view(x.size(0), -1)
+        x = torch.relu(self.fc1(x))
+        x = torch.relu(self.fc2(x))
+        return self.fc3(x)
+
+
+class TestRPS:
+    """Test for RPS."""
+
+    def test_rps_multi(self):
+        """Test for RPS."""
+        # multi-class
+        # init the model
+        model = MLP(num_classes=10)
+        dataset = TensorDataset(torch.randn(10, 1, 28, 28), torch.randint(0, 10, (10,)))
+        full_train_loader = DataLoader(dataset, batch_size=2)
+        test_loader = DataLoader(dataset, batch_size=2)
+        train_loader = DataLoader(
+            dataset,
+            batch_size=2,
+            sampler=SubsetSampler(range(4)),
+        )
+
+        # define the loss function
+        def f(pre_activation_list, label_list):
+            return functional.cross_entropy(pre_activation_list, label_list)
```

Comment:
so we have a somehow different target function than IF/TRAK/TracIN, but I think it's fine.

Reply:
will still name it as `target_func` for now